### PR TITLE
i1661: Allow unidirectional backup targets

### PR DESCRIPTION
**Pull request by MartinEden**
https://vimc.myjetbrains.com/youtrack/issue/VIMC-1661

### CODE REVIEW

File: bin/backup.py [hunk 1]
```diff
@@ -17,4 +17,8 @@ def run_backup():
     targets = list(t for t in settings.targets)
     for target in targets:
         logging.info("- " + target.id)
-        backup_volume(settings, target.mount_id)
+        if target.options.backup:
+            backup_volume(settings, target.mount_id)
+        else:
+            logging.info(f"  (Skipping backing up {target.id} - backup is "
```
Comment (richfitz) [inline bin/backup.py]: f-format strings require python3.6 which is really recent (recent enough that production does not have it for example).  They're nice but possibly not _that_ nice (there is this back though: https://github.com/asottile/future-fstrings).

**Review by richfitz — CHANGES_REQUESTED** [review]
I can't run this without upgrading my python because of the f-string version.  Can we (not necessarily here) think about how to start testing this in general?  This all looks eminently testable...

**Review by hillalex — COMMENTED** [review]
Should there be a corresponding PR in `montagu-bb8`, that updates the `config.json` we use?

File: bin/targets.py [hunk 2]
```diff
@@ -1,10 +1,17 @@
 import docker
 
 
+class TargetOptions:
+    def __init__(self, data):
+        self.backup = data.get("backup", True)
+        self.restore = data.get("restore", True)
```
Comment (hillalex) [inline bin/targets.py:7]: So by default both are set to true. Ok.

**Review by hillalex — APPROVED** [review]
I see that this just sets everything to both backup and restore by default, which seems ok. The update to `montagu-bb8` config can happen in https://vimc.myjetbrains.com/youtrack/issue/VIMC-1663

File: bin/backup.py (hunk 1 above)
Comment (MartinEden) [reply bin/backup.py]: I hadn't realized they were so new. That's a shame, they are nice. I've removed them.

**Comment by MartinEden** [comment]
@littledice: The update to config in montagu-bb8 is the other PR in the the ticket: https://github.com/vimc/montagu-bb8/pull/2

@richfitz: Agreed about tests. I've added this must have ticket to the sprint and will work on it early next week: https://vimc.myjetbrains.com/youtrack/issue/VIMC-1687

**Review by richfitz — APPROVED** [review]

**Pull request merged**